src/screens/AddressList.js

```diff
@@ -72,132 +44,165 @@ class AddressList extends React.Component {
    *
    * @return {Number} Total number of pages of the list
    */
-  getTotalPages = (array) => {
+  const getTotalPages = (array) => {
     return Math.ceil(array.length / WALLET_HISTORY_COUNT);
   }
 
+  useEffect(() => {
+    /* useEffect callback cannot be async, so we create a function to do the asynchronous work inside it
+     * @see https://devtrium.com/posts/async-functions-useeffect#write-the-asynchronous-function-inside-the-useeffect
+     */
+    const fetchAddresses = async () => {
+      const fetchedAddresses = [];
+      const iterator = wallet.getAllAddresses();
+      for (;;) {
+        const addressObj = await iterator.next();
+        const { value, done } = addressObj;
+
+        if (done) {
+          break;
+        }
+
+        fetchedAddresses.push(value);
+      }
+
+      setAddresses(fetchedAddresses);
+      setFilteredAddresses(fetchedAddresses);
+      setTotalPages(getTotalPages(fetchedAddresses));
+    }
+
+    fetchAddresses();
+  }, [wallet]);
```

Comment:
Can `wallet` mutate here? 

If it does, should we cancel the async task?

Maybe something like:

```suggestion
  useEffect(() => {
    let isCancelled = false;
    
    /* useEffect callback cannot be async, so we create a function to do the asynchronous work inside it
     * @see https://devtrium.com/posts/async-functions-useeffect#write-the-asynchronous-function-inside-the-useeffect
     */
    const fetchAddresses = async () => {
      const fetchedAddresses = [];
      const iterator = wallet.getAllAddresses();
      for (;;) {
        if (isCancelled) {
          break;
        }
        const addressObj = await iterator.next();
        const { value, done } = addressObj;

        if (done) {
          break;
        }

        fetchedAddresses.push(value);
      }
      if (!cancelled) {
        setAddresses(fetchedAddresses);
        setFilteredAddresses(fetchedAddresses);
        setTotalPages(getTotalPages(fetchedAddresses));
      }
    }

    fetchAddresses();
    
    return () => {
      isCancelled = true;
    };
  }, [wallet]);
```

Reply:
Good point!
`wallet` cannot mutate, as that would mean that the user has reset the wallet and reloaded it again without ever leaving the `AddressList` screen ( which is absurd ).

Adding the `wallet` property to the effect dependencies was a way to follow the guideline _[Don't lie to React about dependencies](https://overreacted.io/a-complete-guide-to-useeffect/#dont-lie-to-react-about-dependencies)_, but in this case it's not really needed.

So I believe the best course of action would be to remove the dependency altogether and just ignore this guideline as it doesn't fit this screen's business rules.

Reply:
Fixed on [b4481d4](https://github.com/HathorNetwork/hathor-wallet/pull/475/commits/b4481d4b75fcc676ec66b76ad5be3d8aaa57c9b0)